src/Illuminate/Container/Container.php

```diff
@@ -785,6 +785,13 @@ public function build($concrete, $parameters = [])
 			throw new BindingResolutionException($message);
 		}
 
+		if (in_array($concrete, $this->buildStack))
+		{
+			$message = "Cyclic reference found while resolving [$concrete].";
+
+			throw new BindingResolutionException($message);
```

Comment:
Also, I don't think `BindingResolutionException` is really what we want here. We can resolve the binding just fine. We need another exception for this case.